### PR TITLE
Return contiguous tensor from softmax decomposition

### LINKED ISSUE #1135: TIMM model failures:  view size is not compatible with input tensor's size and stride
[Edited]
Repro:
```
benchmarks/timm_models.py -dcuda --inductor --float32 --train --accuracy --only coat_lite_min
benchmarks/timm_models.py -dcuda --inductor --float32 --train --accuracy --only volo_d1_224
```

Error:
```
Traceback (most recent call last):
  File "/fsx/users/binbao/torchdynamo-tip/benchmarks/common.py", line 1210, in run_one_model
    new_result = optimized_model_iter_fn(model, example_inputs)
  File "/fsx/users/binbao/torchdynamo-tip/torchdynamo/eval_frame.py", line 166, in _fn
    return fn(*args, **kwargs)
  File "benchmarks/timm_models.py", line 308, in forward_and_backward_pass
    def forward_and_backward_pass(self, mod, inputs, collect_outputs=True):
  File "benchmarks/timm_models.py", line 309, in <graph break in forward_and_backward_pass>
    cloned_inputs = clone_inputs(inputs)
  File "benchmarks/timm_models.py", line 312, in <graph break in forward_and_backward_pass>
    pred = mod(*cloned_inputs)
  File "/fsx/users/binbao/pytorch-release/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/fsx/users/binbao/conda/envs/release/lib/python3.8/site-packages/timm/models/mlp_mixer.py", line 321, in forward
    def forward(self, x):
  File "/fsx/users/binbao/torchdynamo-tip/torchdynamo/eval_frame.py", line 166, in _fn
    return fn(*args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/aot_autograd.py", line 889, in forward
    return compiled_f(
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/aot_autograd.py", line 875, in new_func
    compiled_fn = create_aot_dispatcher_function(
  File "/fsx/users/binbao/torchdynamo-tip/torchdynamo/utils.py", line 75, in time_wrapper
    r = func(*args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/aot_autograd.py", line 450, in create_aot_dispatcher_function
    aot_dispatch_autograd(flat_fn, fake_flat_tensor_args, aot_config)
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/aot_autograd.py", line 316, in aot_dispatch_autograd
    fx_g = make_fx(functionalize(fake_fn), aot_config.decompositions)(*joint_inputs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/experimental/proxy_tensor.py", line 621, in wrapped
    t = dispatch_trace(wrap_key(func, args, fx_tracer), tracer=fx_tracer, concrete_args=tuple(phs))
  File "/fsx/users/binbao/torchdynamo-tip/torchdynamo/eval_frame.py", line 166, in _fn
    return fn(*args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/experimental/proxy_tensor.py", line 391, in dispatch_trace
    graph = tracer.trace(root, concrete_args)
  File "/fsx/users/binbao/torchdynamo-tip/torchdynamo/eval_frame.py", line 166, in _fn
    return fn(*args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/_symbolic_trace.py", line 739, in trace
    (self.create_arg(fn(*args)),),
  File "/fsx/users/binbao/pytorch-release/torch/fx/_symbolic_trace.py", line 614, in flatten_fn
    tree_out = root_fn(*tree_args)
  File "/fsx/users/binbao/pytorch-release/torch/fx/experimental/proxy_tensor.py", line 405, in wrapped
    out = f(*tensors)
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/eager_transforms.py", line 1420, in wrapped
    func_outputs = func(*func_args, **func_kwargs)
  File "/fsx/users/binbao/pytorch-release/functorch/functorch/_src/aot_autograd.py", line 309, in fake_fn
    return fx_g(primals, tangents)
  File "/fsx/users/binbao/pytorch-release/torch/fx/graph_module.py", line 658, in call_wrapped
    return self._wrapped_call(self, *args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/graph_module.py", line 277, in __call__
    raise e
  File "/fsx/users/binbao/pytorch-release/torch/fx/graph_module.py", line 267, in __call__
    return super(self.cls, obj).__call__(*args, **kwargs)  # type: ignore[misc]
  File "/fsx/users/binbao/pytorch-release/torch/fx/_symbolic_trace.py", line 717, in module_call_wrapper
    return self.call_module(mod, forward, args, kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/experimental/proxy_tensor.py", line 357, in call_module
    return forward(*args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/fx/_symbolic_trace.py", line 710, in forward
    return _orig_module_call(mod, *args, **kwargs)
  File "/fsx/users/binbao/pytorch-release/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "<eval_with_key>.4", line 38, in forward
    view_2 = torch.ops.aten.view.default(getitem_5, [12544, 384]);  getitem_5 = None
  File "/fsx/users/binbao/pytorch-release/torch/_ops.py", line 60, in __call__
    return self._op(*args, **kwargs or {})
RuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.
```

### PR DESCRIPTION
**Pull request by anijain2305**
Fixes https://github.com/pytorch/torchdynamo/issues/1135

Softmax decomp's output stride does not match with aten softmax output stride. Not sure if its desirable. Opening a PR for now.

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -835,6 +835,7 @@ def native_dropout(input: Tensor, p: float, train: Optional[bool]):
 
 @register_decomposition(aten._softmax)
 def _softmax(x: Tensor, dim: int, half_to_float: bool):
+    x = x.contiguous()
```
Comment (ngimel) [inline torch/_decomp/decompositions.py:838]: can you leave a comment here that this is to match eager behavior?

**Review by ngimel — APPROVED** [review]

**Review by eellison — COMMENTED** [review]
https://github.com/pytorch/pytorch/blob/a5cfd8ee5541f4533bc5856915aad630d5e78fa7/torch/_decomp/decompositions.py#L855 as well?

**Review by eellison — COMMENTED** [review]
you may also have the issue in `softmax`/`log_softmax` backward

**Review by ezyang — APPROVED** [review]
Test it?

File: torch/_decomp/decompositions.py (hunk 1 above)
Comment (anijain2305) [reply torch/_decomp/decompositions.py:840]: Done

**Pull request closed without merging**